### ISSUE TITLE
How to use on_dropfile on OS X

### ISSUE THREAD

**Issue by martinsik** [kivy/kivy#3503]
Hey, I can't get working `Window.on_filedrop` event. It looks like the running app isn't recieving files at all. Acording to the [documentation](http://kivy.org/docs/api-kivy.core.window.html#kivy.core.window.WindowBase.on_dropfile) and this [Google Groups](https://groups.google.com/forum/#!topic/kivy-users/pC1AV7Golgc) thread it should work on OS X but it doesn't for me. I've tried filedrop with pure C + SDL2 and it works on my OS X 10.10.3.

```python
import kivy
from kivy.app import App
from kivy.lang import Builder
from kivy.config import Config
from kivy.core.window import Window

kivy.require('1.9.0')
Logger.setLevel('DEBUG')


kv = """
FloatLayout:
    Label:
        text: "Hello, World!"
"""

class SimplePlayerApp(App):
    def build(self):
        Window.bind(on_dropfile=self.file_drop)
        return Builder.load_string(kv)

    def file_drop(self, filename, *args):
        print(filename)

if __name__ == '__main__':
    SimplePlayerApp().run()
```

**Comment by dessant** [comment]
With sdl2 it should work on linux and win too, but this does not seem to work for me:

```python
from kivy.app import App

class DropApp(App):

    def on_start(self, *args):
        self.root_window.bind(on_dropfile=self.file_drop)

    def file_drop(self, *args):
        print(args)

DropApp().run()
```

**Comment by dessant** [comment]
I take that back, it was an atom bug, the print appeared only after closing the app, the above example works.

**Comment by dessant** [comment]
@martinsik, for a start you could try to replace `on_filedrop` with `on_dropfile`. ;)

**Comment by martinsik** [comment]
Ooups :). You're right `on_dropfile`.
Unfortunatelly, it doesn't work either. I fixed my example and tried your example but it still doesn't work.

The app icon doesn't even make any effect when I drag files over it.
Does it really work for you? What version of OS X you have?

```
[INFO   ] [Logger      ] Record log in /Users/martin/.kivy/logs/kivy_15-07-14_38.txt
[INFO   ] [Kivy        ] v1.9.0
[INFO   ] [Python      ] v2.7.6 (default, Sep  9 2014, 15:04:36) 
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.39)]
[INFO   ] [Factory     ] 173 symbols loaded
[INFO   ] [Image       ] Providers: img_tex, img_imageio, img_dds, img_gif, img_sdl2, img_pil (img_ffpyplayer ignored)
[INFO   ] [OSC         ] using <multiprocessing> for socket
[INFO   ] [Window      ] Provider: sdl2
[INFO   ] [GL          ] OpenGL version <2.1 INTEL-10.6.20>
[INFO   ] [GL          ] OpenGL vendor <Intel Inc.>
[INFO   ] [GL          ] OpenGL renderer <Intel HD Graphics 4000 OpenGL Engine>
[INFO   ] [GL          ] OpenGL parsed version: 2, 1
[INFO   ] [GL          ] Shading version <1.20>
[INFO   ] [GL          ] Texture max size <16384>
[INFO   ] [GL          ] Texture max units <16>
[INFO   ] [Window      ] auto add sdl2 input provider
[INFO   ] [Window      ] virtual keyboard not allowed, single mode, not docked
[DEBUG  ] [App         ] Loading kv <./simpleplayer.kv>
[DEBUG  ] [App         ] kv <./simpleplayer.kv> not found
[INFO   ] [Text        ] Provider: sdl2
[DEBUG  ] [Resource    ] add </Library/Fonts> in path list
[DEBUG  ] [Resource    ] add </System/Library/Fonts> in path list
[DEBUG  ] [Resource    ] add </Users/martin/Library/Fonts> in path list
[DEBUG  ] [Resource    ] add </Applications/Kivy.app/Contents/Resources/kivy/kivy/data/fonts> in path list
[DEBUG  ] [Base        ] Create provider from mouse
[INFO   ] [Base        ] Start application main loop
[INFO   ] [GL          ] NPOT texture support is available
[INFO   ] [Base        ] Leaving application in progress...
```

**Comment by dessant** [comment]
@martinsik i have tested on linux only.

**Comment by akshayaurora** [comment]
You need to do some more work on OS X,  including setting up your
info.plist so it can accept files.

Even with that SDL2 version included with kivy does only allow dropping the
files to the icon in launcher,  you will  need to use development version
of SDL2 to get proper dropping of files.

**Comment by martinsik** [comment]
You're probably right, I forgot about the `Info.plist` file. I wanted to try the test app with http://kivy.org/docs/guide/packaging-macosx.html but ran into another problem so I'll post another issue in https://github.com/kivy/kivy-sdk-packager

**Comment by martinsik** [comment]
I tried updating my `Info.plist` and still doesn't work. Tried the code from @dessant, packaged the app with packager and modified plist with:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>CFBundleDevelopmentRegion</key>
	<string>English</string>
	<key>CFBundleDisplayName</key>
	<string>Kivy</string>
	<key>CFBundleDocumentTypes</key>
	<array>
		<dict>
			<key>CFBundleTypeExtensions</key>
			<array>
				<string>mp4</string>
				<string>mpeg4</string>
				<string>m4v</string>
			</array>
			<key>CFBundleTypeName</key>
			<string>MPEG-4 File</string>
			<key>CFBundleTypeRole</key>
			<string>Viewer</string>
		</dict>
	</array>
	<key>CFBundleExecutable</key>
	<string>Kivy</string>
	<key>CFBundleIconFile</key>
	<string>appIcon.icns</string>
	<key>CFBundleIdentifier</key>
	<string>org.kivy.osxlauncher</string>
	<key>CFBundleInfoDictionaryVersion</key>
	<string>6.0</string>
	<key>CFBundleName</key>
	<string>Kivy</string>
	<key>CFBundlePackageType</key>
	<string>APPL</string>
	<key>CFBundleShortVersionString</key>
	<string>stable</string>
	<key>CFBundleVersion</key>
	<string>stable</string>
	<key>LSMinimumSystemVersion</key>
	<string>10.6.0</string>
	<key>LSUIElement</key>
	<true/>
	<key>NSHumanReadableCopyright</key>
	<string>© 2015 qua-non</string>
	<key>NSMainNibFile</key>
	<string>MainMenu</string>
	<key>NSPrincipalClass</key>
	<string>NSApplication</string>
</dict>
</plist>
```

My `CFBundleDocumentTypes` is from VLC player's `Info.plist` that works well on file drop.

**Comment by cbenhagen** [comment]
@martinsik you need sdl2 > 2.0.3 for this to work. No special plist needed.
So what you could do is install the latest sdl2 and all the kivy dependencies via homebrew:
```
brew install --HEAD sdl2
brew install sdl2_image sdl2_ttf sdl2_mixer gstreamer
```
and then install kivy via pip:
```
pip install -I Cython==0.21.2
USE_OSX_FRAMEWORKS=0 pip install git+https://github.com/kivy/kivy.git@1.9.0
```

or you build 2.0.3 with the following patches (untested):
https://hg.libsdl.org/SDL/rev/2cc90bb31777
https://hg.libsdl.org/SDL/rev/63c4d6f1f85f
